neuralteleportation/neuralteleportationmodel.py

```diff
@@ -55,6 +55,26 @@ def initialize_cob(self) -> None:
         """ Set the cob to ones. """
         self.teleport_activations(torch.ones(self.get_cob_size()))
 
+    def init_like_histogram(self, hist: np.ndarray, bin_edges: np.ndarray) -> None:
+        """Initializes the weights of the network as if they were sampled from the provided histogram.
+
+        Args:
+            hist: Probability of each bin in the histogram.
+            bin_edges: Boundaries of each bin (length(hist)+1).
+        """
+        cum_hist = np.cumsum(hist)
+        bin_centers = bin_edges[:-1] + (bin_edges[1] - bin_edges[0])/2
+
+        def init_layer_like_histogram(m):
+            if isinstance(m, nn.Linear):
+                for i in range(m.weight.shape[0]):
+                    for j in range(m.weight.shape[1]):
+                        rand_bin = (np.abs(cum_hist-np.random.rand(1))).argmin()
+                        m.weight[i, j] = bin_centers[rand_bin].item()
```

Comment:
Not quite sure why you added `item()` here? I don't want to get rid of it myself, but I just tested on my end and `bin_centers[rand_bin]` is already a unique float. In that case, `item()` does nothing and simply returns the same number. However, it _does_ lead to confusion, so unless you can justify why you've added it, I'd remove it.

Reply:
If I don't put `item()` then this happens:
```
    m.weight[i, j] = bin_centers[rand_bin]
TypeError: can't assign a numpy.float32 to a torch.FloatTensor
```
But maybe it's just on my side!

Reply:
Yeah, seems weird. I tried running on my end, and `bin_centers[rand_bin]` should be a float, not a tensor. Are you sure your version of the code is from the latest commit?

Reply:
Yes. But actually, I get this error only when I run the `weights_histogram.py` script

Reply:
I was just confused in the end, and searching around a bit I can't seem to find a more intuitive to do the assignment/type conversion, so I think I'll just roll with your solution!